objcgen/EqualsHelper.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.IO;
+
+namespace ObjC {
+	public class EqualsHelper : MethodHelper{
+		public EqualsHelper (TextWriter headers, TextWriter implementation) : base (headers, implementation)
+		{
+			MonoSignature = "Equals (object)";
+			ReturnType = "bool";
+		}
+
+		public void WriteImplementation ()
+		{
+			BeginImplementation ();
+			implementation.WriteLine ("\tif (![other respondsToSelector: @selector (xamarinGetGCHandle)])");
```

Comment:
respondsToSelector will return nil and it will return false.  Double checked in an iOS app with:

```
NSObject *object;
    if (![object respondsToSelector: @selector(foo)])
        NSLog (@"Responds");
```

NSObject is nil and the selector doesn't exist, it hits a breakpoint set on the NSLog